scripts/pr-add-size-label.sh

```diff
@@ -0,0 +1,396 @@
+
+    local var
+
+    for var in "${vars[@]}"
+    do
+        local value
+
+        value=$(printenv "$var" || true)
+
+        [ -n "${value}" ] || die "need to set '$var'"
+    done
+
+    # Force non interactive mode
+    gh config set prompt disabled
+}
+
+# Return an integer representing the "size" of a PR,
+# calculated as the sum of additions and deletions
+# (the amount of change).
+#
+# Note: This function attempts to exclude:
+#
+# - Vendored code changes (pristine upstream files) since vendored code,
+#   by definition, should be treated as "read only" and those changes
+#   result as a _side effect_ of the actual change (which tends to be a
+#   version bump in a configuration file (which causes the pristine vendor
+#   code to be updated).
+#
+# - Auto-generated code.
+#
+# - generated lock/status files for languages like golang and rust.
+get_pr_size()
+{
+    local pr="${1:-}"
+    [ -z "$pr" ] && die "need PR number"
+
+    local stats
+
+    # Determine the amount of change for the specified PR.
+    #
+    # Notes:
+    #
+    # - The exclusions specified here should be similar to the logic in:
+    #
+    #   https://github.com/kata-containers/tests/blob/main/.ci/static-checks.sh
+    #
+    # - Example output showing the diffstat(1) formats:
+    #
+    #     "99 files changed, 12345 insertions(+), 987 deletions(-)"
+    #     "1 file changed, 1 insertion(+)"
+    #     "1 file changed, 2 deletions(-)"
+    stats=$(gh pr diff "$pr" |\
+            filterdiff \
+                --exclude='Cargo.lock' \
+                --exclude='go.mod' \
+                --exclude='go.sum' \
+                --exclude='*.pb.go' \
+                --exclude='*.pb_test.go' \
+                --exclude='*/src/libs/protocols/src/*.rs' \
+                --exclude='*/vendor/*' \
+                --exclude='*/virtcontainers/pkg/cloud-hypervisor/client' \
+                --exclude='*/virtcontainers/pkg/firecracker/client' \
+                |\
+            diffstat -s)
+
+    local additions
+    local deletions
+
+    additions=$(echo "$stats" |\
+        grep -Eo '[0-9][0-9]* insertions?' |\
+        awk '{print $1}' \
+        || echo "0")
+
+    deletions=$(echo "$stats" |\
+        grep -Eo "[0-9][0-9]* deletions?" |\
+        awk '{print $1}' \
+        || echo "0")
+
+    # Validate
+    grep -q "^[0-9][0-9]*$" <<< "$additions" || \
+        die "invalid additions: '$additions'"
+
+    grep -q "^[0-9][0-9]*$" <<< "$deletions" || \
+        die "invalid deletions: '$deletions'"
+
+    local total
+    total=$(( additions + deletions ))
+
+    echo "$total"
+}
+
+# Determine which label to add based on the specified PR size
+get_label_to_add()
+{
+    local size="${1:-}"
+    [ -z "$size" ] && die "need size value"
+
+    local label
+
+    for label in "${!size_ranges[@]}"
+    do
+        local range
+
+        range="${size_ranges[$label]}"
+
+        local value
+
+        if grep -q '^<[0-9][0-9]*$' <<< "$range"
```

Comment:
If let all ranges have the same pattern `min-max`, then there is no need to process by 3 `if-else` branches for example like this https://gist.github.com/liubin/06f01cd347e6a633e2e770c9b809645d

Reply:
True, but although that would simplify that function, the problem is how to handle the maximum. I don't want to have to hard-code something like `500-999999`, so that's why I decided to use `<` and `>`.